.github/workflows/ci_test.yml

```diff
@@ -56,47 +59,72 @@ jobs:
           # we select only a few, in order to conserve resources. If you add
           # an additional environment, please include a comment to indicate
           # why it is useful.
-          - os: ubuntu-latest
+          - name: Black test
+            os: ubuntu-latest
             python: '3.10'
             tox_env: 'black'
             experimental: false
 
           # Basic tests on the oldest & newest supported versions of Python,
           # recording coverage data for the latter.
-          - os: ubuntu-latest
+          - name: Py3.8 with old Astropy version
+            os: ubuntu-latest
             python: '3.8'
             tox_env: 'py38-test-astropy4-cov'
             experimental: false
 
-          - os: ubuntu-latest
+          - name: Linux, Py3.10 with coverage
+            os: ubuntu-latest
             python: '3.10'
             tox_env: 'py310-test-cov'
             experimental: false
 
           # Basic tests on alternative operating systems.
-          - os: windows-latest
+          - name: Windows, Py3.11 with coverage
+            os: windows-latest
             python: '3.11'
             tox_env: 'py311-test-cov'
             experimental: false
 
-          - os: macos-latest
+          - name: Mac OS, Py3.11
+            os: macos-latest
             python: '3.11'
             tox_env: 'py311-test'
             experimental: false
 
           # Test with all optional dependencies installed.
-          - os: ubuntu-latest
+          - name: Linux, Py3.11 all dependencies and coverage
+            os: ubuntu-latest
             python: '3.11'
             tox_env: 'py311-test-alldeps-cov'
             use_remote_data: true
             experimental: false
 
           # Development version of dependencies
-          - os: ubuntu-latest
+          - name: Linux, Py3.11 with dev versions of dependencies
+            os: ubuntu-latest
             python: '3.11'
             tox_env: 'py311-test-devdeps'
             experimental: false
 
+          # Test with all optional dependencies installed.
+          - name: Slow tests on Linux, Py3.11, all deps and coverage
+            os: ubuntu-latest
+            python: '3.11'
+            tox_env: 'py311-test-alldeps-cov'
+            use_remote_data: true
+            experimental: false
+            slow: true
+
+          # Test with all optional dependencies installed.
+          - name: Slow tests on Linux, Py3.11, basic deps and coverage
+            os: ubuntu-latest
+            python: '3.11'
+            tox_env: 'py311-test-cov'
+            use_remote_data: true
+            experimental: false
+            slow: true
```

Comment:
So does this take the previous matrix of operating systems and environments, and basically just writes them out individually? Does this mean we're no longer testing python 3.9, for example, and we're only testing Windows and MacOS on Python 3.11?

Reply:
Our matrix was never an actual matrix, but always a list of single tests, as far as I remember. At this moment, indeed, we are testing with py38, py310 and py311. I couldn't manage to make py312-dev work, so left it out. Would you prefer to also have py39?